test/unit/c_api/eigensolver/test_eigensolver_c_api.cpp

```diff
@@ -0,0 +1,276 @@
+    dlaf_context = dlaf_create_grid(MPI_COMM_WORLD, grid.size().rows(), grid.size().cols(), grid_order);
+  }
+  else if constexpr (api == API::scalapack) {
+#ifdef DLAF_WITH_SCALAPACK
+    // Create BLACS grid
+    Cblacs_get(0, 0, &dlaf_context);  // Default system context
+    Cblacs_gridinit(&dlaf_context, &grid_order, grid.size().rows(), grid.size().cols());
+
+    // Create DLAF grid from BLACS context
+    dlaf_create_grid_from_blacs(dlaf_context);
+#endif
+  }
+
+  // In normal use the runtime is resumed by the C API call
+  // The pika runtime is suspended by dlaf_initialize
+  // Here we need to resume it manually to build the matrices with DLA-Future
+  pika::resume();
+
+  const LocalElementSize size(m, m);
+  const TileElementSize block_size(mb, mb);
+
+  Matrix<const T, Device::CPU> reference = [&]() {
+    auto reference = [&]() -> auto{
+      return Matrix<T, Device::CPU>(GlobalElementSize(m, m), block_size, grid);
+    }
+    ();
+    matrix::util::set_random_hermitian(reference);
+    return reference;
+  }();
+
+  Matrix<T, Device::CPU> mat_a_h(reference.distribution());
+  copy(reference, mat_a_h);
+  mat_a_h.waitLocalTiles();
+
+  eigensolver::EigensolverResult<T, D> ret = [&]() {
+    const SizeType size = mat_a_h.size().rows();
+    Matrix<BaseType<T>, D> eigenvalues(LocalElementSize(size, 1),
+                                       TileElementSize(mat_a_h.blockSize().rows(), 1));
+    Matrix<T, D> eigenvectors(GlobalElementSize(size, size), mat_a_h.blockSize(), grid);
+
+    eigenvalues.waitLocalTiles();
+    eigenvectors.waitLocalTiles();
+
+    char dlaf_uplo = uplo == blas::Uplo::Upper ? 'U' : 'L';
+
+    // Get top left local tiles
+    int lld_a, lld_eigenvectors;
+    T* local_a_ptr;
+    T* local_eigenvectors_ptr;
+    dlaf::BaseType<T>* eigenvalues_ptr;
+    {
+      auto toplefttile_a =
+          pika::this_thread::experimental::sync_wait(mat_a_h.readwrite(LocalTileIndex(0, 0)));
+      auto toplefttile_eigenvalues =
+          pika::this_thread::experimental::sync_wait(eigenvalues.readwrite(LocalTileIndex(0, 0)));
+      auto toplefttile_eigenvectors =
+          pika::this_thread::experimental::sync_wait(eigenvectors.readwrite(LocalTileIndex(0, 0)));
+
+      lld_a = static_cast<int>(toplefttile_a.ld());
+      lld_eigenvectors = static_cast<int>(toplefttile_eigenvectors.ld());
+
+      local_a_ptr = toplefttile_a.ptr();
+      local_eigenvectors_ptr = toplefttile_eigenvectors.ptr();
+      eigenvalues_ptr = toplefttile_eigenvalues.ptr();
+
+    }  // Destroy tiles (avoid spurious dependencies)
+
+    // Suspend pika to ensure it is resumed by the C API
+    pika::suspend();
+
+    if constexpr (api == API::dlaf) {
+      DLAF_descriptor dlaf_desc_a = {(int) m, (int) m, (int) mb, (int) mb, 0, 0, 1, 1, lld_a};
+      DLAF_descriptor dlaf_desc_eigenvectors = {(int) m, (int) m, (int) mb, (int) mb,        0,
+                                                0,       1,       1,        lld_eigenvectors};
```

Comment:
minor: not sure, but you might use `()` instead of `{}` in order to allow narrowing conversions.

Reply:
I tend to use `{}` to avoid narrowing conversion, but I agree in this case doing them explicitly does not make much sense...